lib/datura/elasticsearch/index.rb

```diff
@@ -72,17 +75,26 @@ def get_schema
 
   def get_schema_mapping
     # if mapping has not already been set, get the schema and manipulate
-    if !@schema_mapping
+    if @schema_mapping.nil?
+      @schema_mapping = {}
+      @schema_mapping["dynamic"] = nil
+
       schema = get_schema[@options["es_index"]]
-      @schema_mapping["fields"] = schema["mappings"]["properties"].keys
-      @schema_mapping["dynamic"] = []
-      schema["mappings"]["dynamic_templates"].each do |field_type, info|
-        es_match = info["match"].sub("*", ".*")
-        regex = /^#{es_match}$/
-        @schema_mapping["dynamic"] << info["match"]
+      @schema_mapping["fields"] = schema["mappings"]["_doc"]["properties"].keys
+
+      regex_pieces = []
+      schema["mappings"]["_doc"]["dynamic_templates"].each do |template|
+        mapping = template.map { |k,v| v["match"] }.first
+        # dynamic fields are listed like *_k and will need
+        # to be converted to ^.*_k$, then combined into a mega-regex
+        es_match = mapping.sub("*", ".*")
+        regex = "^#{es_match}$"
+        regex_pieces << regex
+      end
+      if !regex_pieces.empty?
+        regex_joined = regex_pieces.join("|")
+        @schema_mapping["dynamic"] = /#{regex_joined}/
```

Comment:
I would skip the `^` and `$` in each individual regex and just put them on the outside of regex_joined like: `/^(?:#{regex_joined})$/`

Reply:
Good thinking!

Reply:
I tried that and got the following when I ran the tests:
```
   1) Failure:
Datura::ElasticsearchIndexTest#test_get_schema_mapping [test/es_index_test.rb:48]:
--- expected
+++ actual
@@ -1 +1 @@
-/^.*_d$|^.*_i$|^.*_k$|^.*_n$|^.*_t$|^.*_t_en$|^.*_t_es$/
+/^(?:.*_d|.*_i|.*_k|.*_n|.*_t|.*_t_en|.*_t_es)$/
```

Reply:
Actually this may not matter, the test was looking at the literal regex string.